### PR TITLE
fix: remove size from sst metadata

### LINKED ISSUE #321: SstMetaData's size is 0 after flush
### Describe this problem

When test #312, I find size of SstMetaData is 0 after flush memtable, this cause following error when restart(reading sst)

```
2022-10-19 11:49:32.342 ERRO [common_util/src/panic.rs:42] thread 'main' panicked 'Failed to create catalog manager: VisitSysCatalog { source: ReadStream { source: ErrWithSource { msg: "Read record batch", source: ReadFromSubIter { source: PullRecordBat
ch { source: DecodeRecordBatch { source: ExternalError(ParquetError(General("CachableParquetFileReader::get_metadata error: Execution error: file size of 0 is less than footer"))) } } } } } }' at "src/setup.rs:187"
   0: common_util::panic::set_panic_hook::{{closure}}
```

### Steps to reproduce

1. Insert some data
```sql
CREATE TABLE `cpu2` (
`my_ts` timestamp NOT NULL, 
 `t1` string tag,  
 `t2` string tag,  
 `value` double,
  TIMESTAMP KEY(my_ts)) 
  with (
enable_ttl = 'false'
);

insert into cpu2  (my_ts, t1,t2, `value`) 
values
(123, 'web', 'hz', 100),
(124, 'web','hz', 101),
(123, 'ios', 'sh', 100),
(124, 'ios', 'sh', 101);
```
2. flush
```bash
curl --location --request POST 'localhost:5440/flush_memtable'
```

3. restart service

### Expected behavior

No panic

### Additional Information

 memtable is write to sst in
https://github.com/CeresDB/ceresdb/blob/c557007e6c56390d12debe2d69478dcd73ae4796/analytic_engine/src/instance/flush_compaction.rs#L641

but we update size after this, in https://github.com/CeresDB/ceresdb/blob/c557007e6c56390d12debe2d69478dcd73ae4796/analytic_engine/src/instance/flush_compaction.rs#L657
which is too late, since it's already persisted.

### PR DESCRIPTION
**Pull request by jiacai2050**
# Which issue does this PR close?

Closes #321

# Rationale for this change
 

# What changes are included in this PR?

1. Remove size, row_num from sst metadata. Those two fields cannot be known before write sst, so we have to store them in other place(manifest now).
2. When read sst, pass one more arg: file_size, which is stored inside FileHandle.

# Are there any user-facing changes?

No 

# How does this change test

Existing UT

### CODE REVIEW

File: analytic_engine/src/instance/flush_compaction.rs [hunk 1]
```diff
@@ -698,12 +693,15 @@ impl Instance {
         batch_record_senders.clear();
 
         let ret = try_join_all(sst_handlers).await;
```
Comment (Rachelint) [inline analytic_engine/src/instance/flush_compaction.rs:695]:
```suggestion
        let info_and_metas = try_join_all(sst_handlers).await.context(RuntimeJoin)?;
```

File: analytic_engine/src/sst/factory.rs [hunk 2]
```diff
@@ -169,9 +148,10 @@ impl Factory for FactoryImpl {
         options: &SstReaderOptions,
         path: &'a Path,
         store_picker: &'a ObjectStorePickerRef,
+        file_size: u64,
```
Comment (Rachelint) [inline analytic_engine/src/sst/factory.rs:151]: How about defining a struct(for example, `FileContext`?) to wrap `path` and `file_size` ?

File: analytic_engine/src/sst/file.rs [hunk 3]
```diff
@@ -424,6 +424,10 @@ impl FileHandleSet {
 pub struct FileMeta {
     /// Id of the sst file
     pub id: FileId,
+    /// file size in bytes
+    pub size: u64,
+    // total row number
```
Comment (Rachelint) [inline analytic_engine/src/sst/file.rs:429]: /// total row number

File: analytic_engine/src/instance/flush_compaction.rs [hunk 4]
```diff
@@ -979,16 +975,12 @@ impl SpaceStore {
...
-        sst_meta.size = sst_info.file_size as u64;
-
         table_data
             .metrics
-            .compaction_observe_output_sst_size(sst_meta.size);
+            .compaction_observe_output_sst_size(sst_info.file_size as u64);
         table_data
             .metrics
-            .compaction_observe_output_sst_row_num(sst_meta.row_num);
+            .compaction_observe_output_sst_row_num(sst_info.row_num as u64);
```
Comment (Rachelint) [inline analytic_engine/src/instance/flush_compaction.rs:983]: How about
```
let sst_file_size  = sst_info.file_size as u64;
let sst_row_num = sst_info.row_num as u64;
```
And reuse it here and follows?

Comment (jiacai2050) [reply analytic_engine/src/instance/flush_compaction.rs:983]: fixed

File: analytic_engine/src/sst/factory.rs (hunk 2 above)
Comment (jiacai2050) [reply analytic_engine/src/sst/factory.rs:151]: Can't think of any advantage to wrap them together.

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**